# A rental car company claims the mean time to rent a car on their website is 60 seconds with a standard deviation of 30 seconds. A random sample of 36 customers attempted to rent a car on the website. 

## 1) If the mean time to rent was 65 seconds. Is this enough evidence to contradict the company's claim at 5% confidence level? What's the power?

In [41]:
import scipy.stats as stats
import math

In [64]:
# H0: mu - mu0 = 0    H1: mu - mu0 > 0
# It's kind of meaningless to test whether it would take less time to rent a car.

alpha = 0.05
mu = 65
mu0 = 60
n = 36
sigma = 30

z_score = (mu-mu0) / (sigma/math.sqrt(n))
print('z_score: ' + str(z_score))
power = stats.norm.cdf(z_score - stats.norm.ppf(0.95))
print('power: ' + str(round(power, 4)))

z_score: 1.0
power: 0.2595


#### Becase z_score is less than 1.645 (one tail at 95%), we cannot reject the null hypothesis, and the statistical power is 25.95%

## 2) If the analysts team believe the claim is not correct and the the time for rent is 67 seconds, what's the sample size need to designed for a sufficient test with 85% of power

In [70]:
mu0 = 60
mu = 67
sigma = 30
alpha = 0.05
beta = 0.15

n = (sigma * (stats.norm.ppf(1-alpha) + stats.norm.ppf(1-beta)) / (mu-mu0))**2
print('sample size: ' + str(n))

sample size: 132.0483685326968


#### so the sample size would be 133